### PR TITLE
SEC-18177: Enable secret_volumes in adhoc jobs

### PR DESCRIPTION
**Pull request by Molaire**
All this is needed for this to work in paasta local-run is to add it to adhoc_schema.json, conveniant!
Tested locally here: https://yelp.slack.com/archives/C048HCBDD0U/p1690896906993479

I also included a patch about a KeyError because it was trying to grab a non-existent optional key.

### CODE REVIEW

File: paasta_tools/tron_tools.py [hunk 1]
```diff
@@ -418,18 +418,16 @@ def get_action_name(self):
...
-                ),
-                secret_name=secret_volume["secret_name"],
-                container_path=secret_volume["container_path"],
-                default_mode=secret_volume["default_mode"],
-                items=secret_volume["items"],
+                )
             )
-            for secret_volume in secret_volumes
-        ]
+            tron_secret_volume.update(secret_volume)
```
[the remark this answers is not in the document]
Comment (Molaire) [reply paasta_tools/tron_tools.py]: The problem with .get()-ing it is that it creates the key in the TypedDict (with whatever defualt value you want).
The tron validator then see the key and tries to validate it (and would probably fail?)ex: https://sourcegraph.yelpcorp.com/Yelp/Tron/-/blob/tron/config/config_parse.py?L276:5

[the remark this answers is not in the document]
Comment (Molaire) [reply paasta_tools/tron_tools.py]: I don't think having these new args downstream would affect negatively Tron, as we simply don't parse them

[the remark this answers is not in the document]
Comment (Molaire) [reply paasta_tools/tron_tools.py]: Oh no: https://sourcegraph.yelpcorp.com/Yelp/Tron/-/blob/tron/config/config_utils.py?L342-350. OK let me change this

File: paasta_tools/tron_tools.py [hunk 2]
```diff
@@ -415,21 +415,25 @@ def get_job_name(self):
...
                 ),
                 secret_name=secret_volume["secret_name"],
                 container_path=secret_volume["container_path"],
-                default_mode=secret_volume["default_mode"],
-                items=secret_volume["items"],
+                items=secret_volume.get("items", []),
             )
-            for secret_volume in secret_volumes
-        ]
+            if "default_mode" in secret_volume:
```
Comment (nemacysts) [inline paasta_tools/tron_tools.py:432]: would be nice to add a comment for why this is special-cased (we have different places where the default can come from and we don't want to insert the wrong default here)

**Review by nemacysts — APPROVED** [review]

**Pull request merged**